MapboxCoreNavigation/RouteController.swift

```diff
@@ -676,6 +685,30 @@ extension RouteController: CLLocationManagerDelegate {
         }
     }
     
+    func monitorIntersectionProgress(_ location: CLLocation) {
+        guard var intersections = routeProgress.currentLegProgress.currentStepProgress.step.intersections else { return }
+        let currentStepProgress = routeProgress.currentLegProgress.currentStepProgress
+        
+        // The intersections array does not include the upcoming maneuver intersection.
+        if let upcomingStep = routeProgress.currentLegProgress.upComingStep, let upcomingIntersection = upcomingStep.intersections, let firstUpcomingIntersection = upcomingIntersection.first {
+            intersections += [firstUpcomingIntersection]
+        }
+        
+        routeProgress.currentLegProgress.currentStepProgress.intersectionsIncludingUpcomingManeuverIntersection = intersections
+        
+        // Don't search for previous intersections
+        for intersection in intersections.suffix(from: currentStepProgress.intersectionIndex) {
+            let dist = distance(along: currentStepProgress.step.coordinates!, from: location.coordinate, to: intersection.location)
+            // Check if the user is at an intersection
+            if dist <= RouteControllerManeuverZoneRadius {
+                routeProgress.currentLegProgress.currentStepProgress.userIsNearAnIntersection = true
```

Comment:
It should be the job of RouteStepProgress to track how far the user is from an intersection, not a judgment about whether they’re close. It would be more flexible to set `distanceToIntersection` here – or maybe even make that a computed variable on RouteStepProgress – and decide whether it counts as “near” up in `reroutingTolerance`.